lib/DidKeyDriver.js

```diff
@@ -159,52 +179,73 @@ export class DidKeyDriver {
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
    *
    * @param {object} options - Options hashmap.
-   * @param {LDKeyPair|object} options.keyPair - Key used to generate the DID
+   * @param {object} options.keyPair - Key used to generate the DID
    *   document (either an LDKeyPair instance containing public key material,
    *   or a "key description" plain object (such as that generated from a KMS)).
-   *
+   * @param {object} [options.keyAgreementKeyPair] -  Optional
+   *   keyAgreement key pair for generating did for keyAgreement.
    * @returns {Promise<{didDocument: object, keyPairs: Map}>}
    *   Resolves with the generated DID Document, along with the corresponding
    *   key pairs used to generate it (for storage in a KMS).
    */
-  async _keyPairToDidDocument({keyPair} = {}) {
-    const verificationKeyPair =
-      await this.verificationSuite.from({...keyPair});
+  async _keyPairToDidDocument({keyPair, keyAgreementKeyPair} = {}) {
+    let {publicKeyMultibase} = keyPair;
+    if(!publicKeyMultibase && keyPair.publicKeyBase58) {
+      // handle backwards compatibility w/older key pair interfaces
+      publicKeyMultibase = await keyPair.fingerprint();
+    }
+    // get the multibaseMultikeyHeader from the public key value
+    const multibaseMultikeyHeader = getMultibaseMultikeyHeader({
+      value: publicKeyMultibase
+    });
+    const fromMultibase = this._allowedKeyTypes.get(multibaseMultikeyHeader);
+    if(!fromMultibase) {
+      throw new Error(
+        `Unsupported "multibaseMultikeyHeader", "${multibaseMultikeyHeader}".`);
+    }
+    const verificationKeyPair = await fromMultibase({publicKeyMultibase});
+
     const did = getDid({verificationKeyPair});
     verificationKeyPair.controller = did;
     // Now set the source key's id
     setVerificationKeyPairId({verificationKeyPair, did});
-    // get the keyAgreement keypair
-    const {keyAgreementKeyPair, contexts} = await getKeyAgreementKeyPair({
-      verificationKeyPair
+    // get the public components of verification keypair
+    const verificationPublicKey = await verificationKeyPair.export({
+      publicKey: true
     });
+    // get the keyAgreement keypair
+    let contexts;
+    if(!keyAgreementKeyPair) {
```

Comment:
Unless I'm mistaken, this (`contexts`) looks like it's going to be undefined in the multikey case? We need it to have both the DID context and the multikey context in it. Do we have a test that makes sure this is set?

Reply:
Thanks for catching that. Added the multikey context to contexts and added test [da561c8](https://github.com/digitalbazaar/did-method-key/pull/55/commits/da561c82a000a76075ffc0ae6bd6296af528e99c) & [5ce48aa](https://github.com/digitalbazaar/did-method-key/pull/55/commits/5ce48aa3929852582647b584e93ddc73bc36c0b6)